scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,385 @@
+                math.random(100) < 50))
+            then
+                hpp = tempHPP
+                cureTarget = member
+            end
+        end
+
+        if cureTarget then
+            -- curaga can be used on anyone in the alliance and isn't determined by how many need a heal (assumed to be based on skill over cap)
+            if
+                petLvl >= 16 and
+                math.random(100) <= 2 * xi.summon.getSummoningSkillOverCap(pet)
+            then
+                -- curaga
+                if petLvl >= 71 then
+                    spellID = xi.magic.spell.CURAGA_IV
+                elseif petLvl >= 51 then
+                    spellID = xi.magic.spell.CURAGA_III
+                elseif petLvl >= 31 then
+                    spellID = xi.magic.spell.CURAGA_II
+                else
+                    spellID = xi.magic.spell.CURAGA
+                end
+            else
+                if petLvl >= 61 then
+                    spellID = xi.magic.spell.CURE_V
+                elseif petLvl >= 41 then
+                    spellID = xi.magic.spell.CURE_IV
+                elseif petLvl >= 21 then
+                    spellID = xi.magic.spell.CURE_III
+                elseif petLvl >= 11 then
+                    spellID = xi.magic.spell.CURE_II
+                else
+                    spellID = xi.magic.spell.CURE
+                end
+            end
+
+            return spellID, cureTarget
+        else
+            local hasteTier = { 0, 0 }
+            if petLvl >= 40 then
+                hasteTier = { 1, 15 }
+            end
+
+            local regenTier = { 0, 0 }
+            if petLvl >= 21 then
+                regenTier = { 1, 1 }
+            end
+
+            local shellTier = { 0, 0 }
+            if petLvl >= 68 then
+                shellTier = { 4, 2617 }
+            elseif petLvl >= 52 then
+                shellTier = { 3, 2188 }
+            elseif petLvl >= 37 then
+                shellTier = { 2, 1641 }
+            elseif petLvl >= 17 then
+                shellTier = { 1, 1055 }
+            end
+
+            local protectTier = { 0, 0 }
+            if petLvl >= 63 then
+                protectTier = { 4, 140 }
+            elseif petLvl >= 47 then
+                protectTier = { 3, 90 }
+            elseif petLvl >= 27 then
+                protectTier = { 2, 50 }
+            elseif petLvl >= 7 then
+                protectTier = { 1, 20 }
+            end
+
+            for _, member in pairs(party) do
+                local tempDistance = pet:checkDistance(member)
+                if
+                    not member:hasStatusEffect(xi.effect.INVISIBLE) and
+                    (tempDistance <= distance or
+                    not buffTarget)
+                then
+                    -- Light Spirit will overwrite lower tiered Protect and Shell spells but will not overwrite Haste or Regen
+                    local tempSpellID = 0
+                    if
```

Comment:
This block looks suspiciously like it can be written as a table+driver

Reply:
Done. Hopefully you like it. I'm pretty happy of the results